Util/include/Poco/Util/ServerApplication.h

```diff
@@ -122,8 +123,21 @@ class Util_API ServerApplication: public Application
 	/// --pidfile=/var/run/sample.pid) may be useful to record the process ID of
 	/// the daemon in a file. The PID file will be removed when the daemon process
 	/// terminates (but not, if it crashes).
+	/// 
+	/// An application can register a callback to be called at termination time.
+	/// An example of the termination callback registration at some point
+	/// during the ServerApplication initialization time:
+	/// 
+	///     static const std::string cbMsg = "Termination custom message"s;
+	///     auto tCB = [](const std::string& message)
+	///     {
+	///         std::cout << message << std::endl;
+	///     };
+	///     ServerApplication::registerTerminateCallback(tCB, cbMsg);
 {
 public:
+	using TerminateCallback = std::function<void(const std::string&)>;
```

Comment:
What is the purpose of the `std::string` argument?

Reply:
I guess whatever is needed in there can just be captured by lambda